conda/install.py

```diff
@@ -569,18 +635,56 @@ def extract(pkgs_dir, dist):
                 for fn in files:
                     p = join(root, fn)
                     os.lchown(p, 0, 0)
+        add_cached_package(pkgs_dir, rec['urls'][0], overwrite=True)
+        if path not in rec['dirs']:
+            rec['dirs'].append(path)
 
-def is_extracted(pkgs_dir, dist):
-    return (isfile(join(pkgs_dir, dist, 'info', 'files')) and
-            isfile(join(pkgs_dir, dist, 'info', 'index.json')))
+def is_extracted(dist):
+    for fn in package_cache().get(dist, {}).get('dirs', ()):
+        return fn
+    return None
 
-def rm_extracted(pkgs_dir, dist):
-    with Locked(pkgs_dir):
-        path = join(pkgs_dir, dist)
-        rm_rf(path)
+def rm_extracted(dist):
+    rec = package_cache().get(dist)
+    if rec is None:
+        return
+    for fname in rec['dirs']:
+        with Locked(dirname(fname)):
+            rm_rf(fname)
+    rec['dirs'] = []
+
+def find_new_location(dist):
+    rec = package_cache().get(dist)
+    if rec:
+        return dirname((rec['files'] or rec['dirs'])[0]), None
```

Comment:
I think here you might need:
``
return dirname(rec['files'][0] if rec['files'][0] else rec['dirs'][0]), None
``
Since rec['files'] and rec['dirs'] are lists containing single None entries in the test I ran.